mobile/src/modules/home/components/HomeStackedAreaChart.tsx

```diff
@@ -0,0 +1,159 @@
+
+  state = {
+    selectedDateRange: 100,
+    priceInBTC: false
+  };
+
+  componentWillMount () {
+    // this.props.dispatch(updateAccountTransactionForMultipleAccounts(this.props.accounts));
+  }
+
+  calculateChartData () {
+    console.log(this.props.priceApi.historicalPriceInfo);
+    const accountTransactions = this.props.accounts.map((account) => {
+      if (account.transactions) {
+        return getBalanceHistoryFromTransactions(account.account,
+          convertNQTStringToNumber(account.balanceNQT), account.transactions);
+      }
+    });
+    const now = new Date();
+    const data = [];
+    const historicalPrice: Metric[] = [];
+    let i = 0;
+    for (const d = new Date(new Date().setDate(now.getDate() - this.state.selectedDateRange));
+      d <= now; d.setDate(d.getDate() + 1)) {
+      data[i] = {
+        day: new Date(d)
+      };
+
+      if (this.state.priceInBTC) {
+        const price = this.props.priceApi.historicalPriceInfo.Data.find((metric) => {
+          return this.sameDay(new Date(metric.time * 1000), d);
+        });
+        historicalPrice[d.getTime()] = price || this.props.priceApi.historicalPriceInfo.Data[0];
+      }
+      accountTransactions.map((transactions, accountIndex) => {
+        // set to 0 by default, in case we can't match a transaction
+        data[i][`account${accountIndex}`] = 0;
+
+        if (transactions && transactions.length) {
+          for (let t = transactions.length - 1; t >= 0; t--) {
+            const transaction = transactions[t];
+            // only match up to the most recent transaction, not beyond
+            if (convertBurstTimeToDate(transaction.timestamp) > d) {
+              continue;
+            }
+            if (this.state.priceInBTC) {
+              const priceInBTCOnThatDay = historicalPrice[d.getTime()];
+              data[i][`account${accountIndex}`] = transaction.balance * priceInBTCOnThatDay.close;
+            } else {
+              data[i][`account${accountIndex}`] = transaction.balance;
+            }
+          }
+        }
+      });
+      i++;
+    }
+
+    return data;
+  }
+
+  toggleBTCPrice = () => {
+    this.setState({ priceInBTC: !this.state.priceInBTC });
+  }
+
+  sameDay (d1, d2) {
+    return d1.getFullYear() === d2.getFullYear() &&
+      d1.getMonth() === d2.getMonth() &&
+      d1.getDate() === d2.getDate();
+  }
+
+  render () {
+
+    const data = this.calculateChartData();
+    // todo: make colors and counts dynamic
+    const colors = ['rgba(171,171,171,1)', 'rgba(37,37,37,1)', 'rgba(255, 255, 255, 1)', 'rgba(51, 109, 181, 1)'];
+    const keys = Object.keys(data[0]).slice(1); // remove 'day' from the keys to get the account names
+
+    const svgs = [
+      { onPress: () => console.log('account0') },
+      { onPress: () => console.log('bananas') }
+    ];
+
+    return (
+      <>
+        <StackedAreaChart
+          style={{ height: 200, paddingVertical: 16, zIndex: 1 }}
```

Comment:
Styles here and below should be extracted to const outside of component. For dynamic things we can pass an array, like `style={[styles.chart, disabled && styles.disabledChart]}`